debezium-server/debezium-server-eventhubs/src/main/java/io/debezium/server/eventhubs/EventHubsChangeConsumer.java

```diff
@@ -125,57 +125,80 @@ public void handleBatch(List<ChangeEvent<Object, Object>> records,
             op.setMaximumSizeInBytes(maxBatchSize);
         }
 
-        EventDataBatch batch = producer.createBatch(op);
+        for (int i = 0; i < records.size();) {
+            int added = 0;
+            int start = i;
+            LOGGER.trace("Emitting events starting from index {}", start);
+
+            EventDataBatch batch = producer.createBatch(op);
+
+            // this loop adds as many records to the batch as possible
+            for (; i < records.size();) {
+                ChangeEvent<Object, Object> record = records.get(i);
+                LOGGER.trace("Received record '{}'", record.value());
+                if (null == record.value()) {
+                    continue;
```

Comment:
The position in the Debezium buffer is not incremented here. Will not the code end in endless loop?

Reply:
Nice spot, you're absolutely correct; I'll fix!

Reply:
@Naros, is this still  pening?

Reply:
@gunnarmorling Nope that was fixed in commit b069aef3eb39f8d3fc93408e529fbac246bd5a56